### ISSUE TITLE
Clean up testgrid reports

### ISSUE THREAD

**Issue by jose-cortina** [kyma-project/test-infra#2492]
**Description**

https://testgrid.k8s.io/kyma-all shows a number of old jobs which are no longer executed. This causes confusion and lack of transparency on the tests we actually have.

It would be good to clean up the list of jobs that appear there and have only the ones that are still relevant.

**Comment by adamwalach** [comment]
The definition for testgrid is stored there: https://github.com/kubernetes/test-infra/tree/master/config/testgrids/kyma

**Comment by Ressetkk** [comment]
## Maintain and automation
I came up with an idea of how to handle the problem and make it easier to maintain in the future. TestGrid provides great ability to create configuration based on *annotations* fields present in prowjobs definition. According to the TestGrid [Configuration doc](https://github.com/GoogleCloudPlatform/testgrid/blob/master/config.md) the YAML `annotations` field allows to define configuration of each prowjob how it behaves in the TestGrid. It is possible to define:
* on which dashboard and tab the job should be present
* email alerts
* failure conditions to alert
* and much more. 

Additionally k8s' test-infra provides a script called [Transfigure.sh](https://github.com/kubernetes/test-infra/tree/master/testgrid/cmd/transfigure) which automatically generates TestGrid configuration and creates PR with a new config to kubernetes/test-infra. There is prebuilt image with the script and example prowjobs. The same approach is implemented by [istio](https://prow.istio.io/?job=post-istio-test-infra-upload-testgrid-config). All of their jobs define `annotations` field with the configuration of TestGrid.

With this approach the TestGrid will be up-to-date with current prow config without additional manual steps apart from adding configuration in a prowjob.

## Cleaning up
As we are having two different organisations for Kyma - *kyma-project* and *kyma-incubator* the best approach for that would be to define two different dashboards for each of the organisation. The jobs for each organisation would be grouped in tabs with the name of a component or project they are testing. For example all jobs related to console should be in `Kyma` dashboard under `console-tests` tab. 
As an exception all integration jobs can be placed in `Kyma` dashboard under `integration-tests` tab.

That should give better insights on the running jobs. 

If this approach is fine then all of the jobs in test-infra repository will have to be updated with the testgrid configuration.

**Comment by Ressetkk** [comment]
We introduced an automation for generating TestGrid reports by using annotations. The updated documentation on how to add the job to the TestGrid is available here https://github.com/kyma-project/test-infra/blob/master/docs/prow/prow-k8s-testgrid.md. Additionally to get all the functionalities we had (60 days of results) I've created an external contribution which adds the ability to define it using annotations https://github.com/kubernetes/test-infra/pull/18408. A new config has been updated and the result is present on https://testgrid.k8s.io
